### PR TITLE
Issues 4, 5, and add `da00`

### LINKED ISSUE #4: The regex pattern `\d` is interpreted as an escape sequence
Python has for some time issued a `DeprecationWarning` when encountering any backslash-character pair  in a string that is not a valid escape sequence. [As of Python 3.12 that is changed to a `SyntaxWarning` and will be replaced in the future by a `SyntaxError](https://docs.python.org/3/whatsnew/3.12.html#other-language-changes).

`wherefore` uses the regular expression pattern `\d` at a number of places:
https://github.com/ess-dmsc/wherefore/blob/65e77d3289c2835e81eab7b90a3072918e88ff57/wherefore.py#L16
https://github.com/ess-dmsc/wherefore/blob/65e77d3289c2835e81eab7b90a3072918e88ff57/wherefore.py#L26
https://github.com/ess-dmsc/wherefore/blob/65e77d3289c2835e81eab7b90a3072918e88ff57/wherefore.py#L37
https://github.com/ess-dmsc/wherefore/blob/65e77d3289c2835e81eab7b90a3072918e88ff57/wherefore.py#L43

Per the linked Python change log, these regular expression strings should be replaced by raw-strings to remove the `SyntaxWarning`.

### LINKED ISSUE #5: New data-streaming-types not recognized
The hard-coded list of 'type extractors' and logic flow of `extract_message_info` ensures that new streaming data types are reported as `Unknown | Unknown` when examined with `wherefore`.

Since the schema type is extracted, its value can be reported even if no other information can be decoded due to, e.g., a missing deserialiser in `python-streaming-data-types`.

### PR DESCRIPTION
**Pull request by g5t**
Fixes issue #4 by changing regex strings with illegal escape characters into raw strings.

Fixes issue #5 by adding an 'extractor' for unknown streaming data types that reports its name such that, e.g., `ar51` messages are identified when examining their topic.

Adds an extractor for `da00`.

### CODE REVIEW

File: wherefore/Message.py [hunk 1]
```diff
@@ -304,17 +313,22 @@ def se00_extractor(data: bytes) -> Tuple[str, Optional[datetime], str]:
     "f144": f144_extractor,
     "se00": se00_extractor,
     "al00": al00_extractor,
+    "da00": da00_extractor,
 }
 
 
+def unknown_extractor(name: str):
+    def named_extractor(data: bytes) -> Tuple[str, datetime | None, str]:
```
Comment (hurvan) [inline wherefore/Message.py]: Use Tuple[str, Union[datetime, None], str] instead

File: wherefore/Message.py [hunk 2]
```diff
@@ -279,6 +280,14 @@ def se00_extractor(data: bytes) -> Tuple[str, Optional[datetime], str]:
         f"Nr of elements: {len(message.values)}",
     )
 
+def da00_extractor(data: bytes) -> Tuple[str, datetime | None, str]:
```
Comment (hurvan) [inline wherefore/Message.py]: Use Tuple[str, Union[datetime, None], str] instead

**Review by hurvan — COMMENTED** [review]
LGTM and tested it! But use Union instead of "|" for the environments that are less than python3.10

**Comment by g5t** [comment]
> LGTM and tested it! But use Union instead of "|" for the environments that are less than python3.10

Good catch, I've switched out the bitwise or syntax for `Optional[datetime]` to match the other extractor signatures.

**Review by hurvan — APPROVED** [review]
LGTM

**Pull request merged**